crates/utils/re_depth_compression/src/ros_rvl.rs

```diff
@@ -0,0 +1,375 @@
+//! RVL (Range Image Visualization Library) codec helpers.
+
+use std::mem::size_of;
+
+use byteorder::{ByteOrder as _, LittleEndian};
+use thiserror::Error;
+
+const CONFIG_HEADER_SIZE: usize = size_of::<i32>() + size_of::<[f32; 2]>();
+const RESOLUTION_HEADER_SIZE: usize = size_of::<[u32; 2]>();
+
+/// Metadata extracted from a ROS2 `compressedDepth` RVL payload.
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub struct RvlMetadata {
+    pub width: u32,
+    pub height: u32,
+    pub depth_quant_a: f32,
+    pub depth_quant_b: f32,
+    payload_offset: usize,
+    num_pixels: usize,
+}
+
+impl RvlMetadata {
+    #[inline]
+    pub fn payload<'a>(&self, bytes: &'a [u8]) -> Result<&'a [u8], RvlDecodeError> {
+        bytes
+            .get(self.payload_offset..)
+            .ok_or(RvlDecodeError::UnexpectedEof)
+    }
+
+    #[inline]
+    pub fn num_pixels(&self) -> usize {
+        self.num_pixels
+    }
+}
+
+#[derive(Debug, Error)]
+pub enum RvlDecodeError {
+    #[error("compressed depth payload missing RVL header")]
+    MissingHeader,
+
+    #[error("RVL payload missing resolution header")]
+    MissingResolution,
+
+    #[error("RVL payload reports zero resolution")]
+    ZeroResolution,
+
+    #[error("RVL image resolution would overflow")]
+    ResolutionOverflow,
+
+    #[error("RVL payload shorter than expected for resolution {width}x{height}")]
+    PayloadLengthMismatch { width: u32, height: u32 },
+
+    #[error("RVL stream encodes more zero pixels than expected")]
+    TooManyZeros,
+
+    #[error("RVL stream encodes more non-zero pixels than expected")]
+    TooManyNonZeros,
+
+    #[error("RVL stream encoded an empty run (zero zeros and zero non-zeros)")]
+    NoProgress,
+
+    #[error("RVL decoded value {value} does not fit into u16")]
+    ValueOutOfRange { value: i32 },
+
+    #[error("RVL stream ended unexpectedly")]
+    UnexpectedEof,
+
+    #[error("RVL VLE value overflowed")]
+    ValueOverflow,
+}
+
+pub fn parse_ros_rvl_metadata(data: &[u8]) -> Result<RvlMetadata, RvlDecodeError> {
```

Comment:
could use some doc string. I tried to read up on this ros specific (?) metadata but couldn't find anything